frontend/src/pages/CollaborationReportForm/Pages/components/NextStepsRepeater.js

```diff
@@ -0,0 +1,199 @@
+
+import './NextStepsRepeater.scss';
+import ControlledDatePicker from '../../../../components/ControlledDatePicker';
+import Req from '../../../../components/Req';
+import PlusButton from '../../../../components/GoalForm/PlusButton';
+import { isValidDate } from '../../../../utils';
+
+const DEFAULT_STEP_HEIGHT = 80;
+
+export default function NextStepsRepeater({
+  name,
+  ariaName,
+  required,
+}) {
+  const [heights, setHeights] = useState([]);
+
+  const {
+    register,
+    control,
+    getValues,
+    errors,
+    setError,
+  } = useFormContext();
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name,
+    keyName: 'key', // because 'id' is the default key switch it to use 'key'.
+  });
+
+  const canDelete = fields.length > 1;
+
+  const onRemoveStep = (index) => {
+    // Remove from Array.
+    remove(index);
+
+    // Remove Height.
+    const updatedHeights = [...heights];
+    updatedHeights.splice(index, 1);
+    setHeights(updatedHeights);
+  };
+
+  // istanbul ignore next - too hard to test heights in jsdom
+  const onStepTextChanged = (e, index) => {
+    // Adjust Text Area Height If Greater than Default Height.
+    const existingHeights = [...heights];
+    if (e.target && e.target.scrollHeight && e.target.scrollHeight > DEFAULT_STEP_HEIGHT) {
+      existingHeights[index] = `${e.target.scrollHeight}px`;
+      setHeights(existingHeights);
+    }
+  };
+
+  const onAddNewStep = () => {
+    const allValues = getValues();
+    const fieldArray = allValues[name] || [];
+    const canAdd = fieldArray.every((field, index) => {
+      if (!field.collabStepDetail) {
+        setError(`${name}[${index}].collabStepDetail`, { message: 'Please enter a next step' });
+      }
+
+      const parsedDate = isValidDate(field.collabStepCompleteDate);
+      if (!field.collabStepCompleteDate || !parsedDate) {
+        setError(`${name}[${index}].collabStepCompleteDate`, { message: 'Please enter a valid date' });
+      }
+
+      const isValid = !(
+        errors[name]?.[index]?.collabStepDetail
+        || errors[name]?.[index]?.collabStepCompleteDate
+      );
+
+      return isValid;
+    });
+    if (canAdd) {
+      append({ id: null, collabStepDetail: '', collabStepCompleteDate: null });
+    }
+  };
+
+  const dateLabel = () => 'When do you anticipate completing this step?';
+
+  const textareaRegister = (() => {
+    if (required) {
+      return register({ required: 'Enter a next step' });
+    }
+    return register();
+  })();
+
+  return (
+    <>
+      <div className="ttahub-next-steps-repeater">
+        {fields.map((item, index) => (
+          <div key={item.key}>
+            <FormGroup
+              className="margin-top-2 margin-bottom-2"
+              error={(errors[name] && errors[name][index]
+                && errors[name][index].collabStepDetail)}
+            >
+              <Label
+                htmlFor={`$next-step-${index + 1}`}
```

Comment:
```suggestion
                htmlFor={`next-step-${index + 1}`}
```

Reply:
@Andrew565  maybe obvious note but if you change this here as Krys suggests, you also need to change the associated ID so the label is properly connected to the input